content/Watsonx/WatsonxAI/files/204/use_case_rag.ipynb

```diff
@@ -0,0 +1,313 @@
+    "them to answer question based on the information in the document.\n",
+    "\n",
+    "To get started we'll first verify that you have the necessary dependencies installed to run this notebook.\n",
+    "\n",
+    "Go ahead and run the following code cell."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Install dependencies\n",
+    "import sys\n",
+    "!{sys.executable} -m pip install -q chromadb==0.4.22\n",
+    "!{sys.executable} -m pip install -q ibm_watson_machine_learning==1.0.342\n",
+    "!{sys.executable} -m pip install -q langchain==0.1.4\n",
+    "!{sys.executable} -m pip install -q langchain_community==0.0.15\n",
+    "!{sys.executable} -m pip install -q pypdf==4.0.1\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Bring in dependencies\n",
+    "\n",
+    "In this next code cell we'll bring in all the dependencies we'll need for later use.\n",
+    "\n",
+    "Go ahead and run the following code cell. **There should be no ouput**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Bring in dependencies\n",
+    "# SQLite fix: https://docs.trychroma.com/troubleshooting#sqlite\n",
+    "# __import__('pysqlite3')\n",
+    "# import sys\n",
+    "# sys.modules['sqlite3'] = sys.modules.pop('pysqlite3')\n",
+    "\n",
+    "import requests\n",
+    "import chromadb\n",
+    "from langchain.text_splitter import RecursiveCharacterTextSplitter\n",
+    "from chromadb.utils import embedding_functions\n",
+    "\n",
+    "# Document loaders\n",
+    "from langchain.document_loaders.pdf import PyPDFLoader\n",
+    "from langchain.document_loaders import TextLoader\n",
+    "from langchain_core.documents import Document\n",
+    "\n",
+    "# WML python SDK\n",
+    "from ibm_watson_machine_learning.foundation_models import Model\n",
+    "from ibm_watson_machine_learning.metanames import GenTextParamsMetaNames as GenParams\n",
+    "from ibm_watson_machine_learning.foundation_models.utils.enums import DecodingMethods\n",
+    "\n",
+    "print(\"Successfully loaded dependencies!\")\n",
+    "\n",
+    "FILE_TYPE_TXT = \"txt\"\n",
+    "FILE_TYPE_PDF = \"pdf\"\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Some important variables\n",
+    "\n",
+    "In this next code cell you'll define some variables that will be used in order to interact with your instance of watsonx.ai.\n",
+    "\n",
+    "Go ahead and run the following code cell. **There should be no ouput**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Update the global variables that will be used for authentication in another function\n",
+    "watsonx_project_id = \"PASTE_PROJECT_ID_HERE\"\n",
```

Comment:
Should include steps for where to retrieve Project ID and API Key

Reply:
We have [this issue](https://github.com/ibm-build-lab/VAD-VAR-Workshop/issues/315) to create a reusable component since we reference this in multiple places.

Reply:
Perfect. We should reference that here once that issue is done and that component is created